Cargo.toml

```diff
@@ -86,15 +86,35 @@ proptest-derive = { version = "0.7", default-features = false }
 serde = { version = "1.0", default-features = false, features = ["alloc", "derive", "rc"] }
 serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
 serde-untagged = {version = "0.1" }
-sha2 = { version = "0.10", default-features = false }
 smallvec = { version = "1.15", default-features = false, features = ["union", "const_generics", "const_new"] }
 thiserror = { version = "2.0", default-features = false }
 tokio = { version = "1.48", default-features = false }
 tracing = { version = "0.1", default-features = false, features = ["attributes"] }
-winter-air = { version = "0.13", default-features = false }
-winter-fri = "0.13"
-winter-math = { version = "0.13", default-features = false }
-winter-prover = { version = "0.13", default-features = false }
-winter-rand-utils = "0.13"
-winter-utils = { version = "0.13", default-features = false }
-winter-verifier = { version = "0.13", default-features = false }
+rand = { version = "0.9", default-features = false }
+sha2 = { version = "0.10", default-features = false }
+
+# Plonky3 dependencies
+p3-air        = { branch = "zz/migrate-plonky3", default-features = false, git = "https://github.com/0xMiden/Plonky3/" }
```

Comment:
`maybe-rayon` has the `parallel` feature that we may want to enable. All of these crates re-export it, but since they all depend on `maybe-rayon`, setting the feature only once there is enough to enable it for all upstream crates, since Rust will only compile it once.

All of this to say, we might want to add a `parallel = ["p3-maybe-rayon/parallel"]` feature, or add it to our `concurrent` one.